src/rate_limiter/src/persist.rs

```diff
@@ -72,10 +75,12 @@ impl Persist<'_> for RateLimiter {
     fn restore(_: Self::ConstructorArgs, state: &Self::State) -> Result<Self, Self::Error> {
         let rate_limiter = RateLimiter {
             // Safe to unwrap because TokenBucket::restore doesn't return errors.
-            ops: state
-                .ops
-                .as_ref()
-                .map(|ops| TokenBucket::restore((), ops).unwrap()),
+            ops: state.ops.as_ref().map(|ops| {
+                TokenBucket::restore((), ops).unwrap_or_else(|e| {
+                    error!("Failed to restore rate limiter: {}", e);
+                    std::process::exit(i32::from(EXIT_CODE_GENERIC_ERROR));
```

Comment:
Why exit the process here instead of propagating the error? AFAIK we're not doing this in the other restore contexts.

Reply:
Indeed, as a general rule, if the _current_ function returns a `Result` you should propagate any local errors through that Result.
Exceptions to this rule are error cases where running further instructions is either not safe or doesn't make sense from a program logic point of view (for example: mutex lock is poisoned - continuing the program generally results in Undefined Behavior).

Reply:
Good point. I'll change that.